forest/daemon/src/daemon.rs

```diff
@@ -398,18 +391,51 @@ pub(super) async fn start(config: Config, detached: bool) {
     info!("Forest finish shutdown");
 }
 
+async fn prompt_and_fetch_snapshot(config: &mut Config) {
+    if !config.client.download_snapshot {
+        if let false = Confirm::with_theme(&ColorfulTheme::default())
+            .with_prompt(
+                "Forest needs a snapshot to sync with the network. Would you like to download one now?",
+            )
+            .default(false)
+            .interact()
+            .expect("infallible. yes or no with default")
+        {
+            cli_error_and_die(
+                "Forest cannot sync without a snapshot. Download a snapshot from a trusted source and import with --import-snapshot=[file]",
+                1
+            );
+        }
+    }
+
+    let snapshot_path = config
+        .client
+        .data_dir
+        .join("snapshots")
+        .join(config.chain.name.clone());
+
+    match snapshot_fetch(&snapshot_path, config).await {
+        Ok(snapshot_path) => {
+            config.client.snapshot_path = Some(snapshot_path);
+            config.client.snapshot = true;
+        }
+        Err(e) => cli_error_and_die(e.to_string(), 1),
+    };
+}
+
 async fn sync_from_snapshot(config: &Config, state_manager: &Arc<StateManager<RocksDb>>) {
     if let Some(path) = &config.client.snapshot_path {
         let stopwatch = time::Instant::now();
         let validate_height = if config.client.snapshot {
             config.client.snapshot_height
         } else {
+            info!("SNAPSHOT_HEIGHT WAS 0");
```

Comment:
This looks like a left-over debug statement.

Reply:
it is